server/circleci.go

```diff
@@ -20,42 +20,38 @@ import (
 	"github.com/mattermost/go-circleci"
 )
 
-func (s *Server) triggerCircleCiIfNeeded(ctx context.Context, pr *model.PullRequest) {
+func (s *Server) triggerCircleCiIfNeeded(ctx context.Context, pr *model.PullRequest) error {
 	mlog.Info("Checking if need trigger circleci", mlog.String("repo", pr.RepoName), mlog.Int("pr", pr.Number), mlog.String("fullname", pr.FullName))
 	repoInfo := strings.Split(pr.FullName, "/")
 	if repoInfo[0] == s.Config.Org {
 		// It is from upstream mattermost repo don't need to trigger the circleci because org members
 		// have permissions
-		mlog.Info("Don't need to trigger circleci", mlog.String("repo", pr.RepoName), mlog.Int("pr", pr.Number), mlog.String("fullname", pr.FullName))
-		return
+		return nil
 	}
 
 	// Checking if the repo have circleci setup
 	builds, err := s.CircleCiClient.ListRecentBuildsForProjectWithContext(ctx, circleci.VcsTypeGithub, pr.RepoOwner, pr.RepoName, "master", "", 5, 0)
 	if err != nil {
-		mlog.Error("listing the circleci project", mlog.String("repo", pr.RepoName), mlog.Int("pr", pr.Number), mlog.String("Fullname", pr.FullName), mlog.Err(err))
-		return
+		return fmt.Errorf("could not list the circleci builds for project: %w", err)
```

Comment:
Do we need to wrap the error? AFAIK we need to wrap errors if we want to check with `errors.Is` or something like that but maybe I'm wrong

Reply:
I believe it's a good practice to pass errors upwards. Even if we don't unwrap it or check for specific error at caller site, we are chaining the errors and get a nice trace.

Reply:
Depends. I read about it [here](https://blog.golang.org/go1.13-errors) in the "Whether to wrap" section. I thought that is a good practice to do it always but is not that way as it looks. Just wanted to give a heads up about it :D